cmds/clients/contestcli-http/main.go

```diff
@@ -149,12 +183,40 @@ func run(verb string) error {
 		encoder.SetIndent("", " ")
 		err := encoder.Encode(apiErr)
 		if err != nil {
-			return fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
+			return "", fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
 		}
 		indentedJSON = buffer.Bytes()
 	}
 	// this is the only thing we want on stdout - the JSON-formatted response,
 	// so it can be piped to other tools if desired.
-	fmt.Println(string(indentedJSON))
-	return nil
+	return string(indentedJSON), nil
 }
+ 
+func wait(params url.Values) (string, error) {
+	
+	var finished bool
+	time.Sleep(jobWaitPoll)
+	
+	resp, err := request("status", params)
+	if err != nil {
+		return "", err
+	}
+	
+	var parsedResp map[string]interface{}
+	if err := json.Unmarshal([]byte(resp), &parsedResp); err != nil {
+		return "", fmt.Errorf("cannot decode json respnse: %v", err)
+	}
+	jobState := parsedResp["Data"].(map[string]interface{})["Status"].(map[string]interface{})["State"]
+
+	for _, eventName := range jobmanager.JobCompletionEvents {
+		if fmt.Sprintf("%v", jobState) == fmt.Sprintf("%v", eventName) {
```

Comment:
`event.Name` is aliased on `string`, so you can just use a type conversion, to eliminate the overhead of a function call (I believe that Sprintf is not optimized out here, while the type conversion should be done at compile time):
```
if string(jobState) == string(eventName) {
```

Reply:
Done